earth_enterprise/src/fusion/portableglobe/cutter/cgi-bin/core/glc_assembler.py

```diff
@@ -172,12 +178,18 @@ def PrepareAssembler(self, base_path):
     utils.PrintAndLog("Prepare assembler done.", logger)
     return logger
 
-  def GetLogger(self, base_path):
+  def GetLogger(self):
     """Get logger built from info.txt."""
-    info_file = INFO_FILE % base_path
-    if not os.path.exists(os.path.dirname(info_file)):
-      os.makedirs(os.path.dirname(info_file))
-    logger = utils.Log(info_file)
+
+    if not self.info_file:
+      # if called before setting self.info_file
+      # return a logger that does nothing
+      logger = utils.Log("/nulllogger", False)
```

Comment:
why would we need this ? a logger that does nothing

Reply:
Shouldn't happen but if it did I think a logger that does nothing is better than a run-time exception that will likely happen if I returned None.  Or if I don't return None then where do I put the file?  Again should not happen but if it does I feel a null logger is the least bad of all bad options available.

Reply:
this would not happen because we initialize the info file to an empty string ? (self.info_file = "") ? if yes, i would remove this code as it might be confusing.

Reply:
I added the initialization just to be sure the attribute existed.  But it still needs to be valued with something other than empty string.

Reply:
I don't see the point of the code, either.  The purpose of exceptions is to report errors.  If there's an error you should throw an exception.  All the better if the exception is already thrown for you, and you don't need to write any code for it.